### PR TITLE
UX: suspend forever time period messages

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
When the Forever option is selected for suspending a user, the user is suspended for 1000 years. Without customizing the site’s text, this time period is displayed to the user in the suspension email that is sent to the user, and if the user attempts to log back into the site. Telling someone that they have been suspended for 1000 years seems likely to come across as a bad attempt at humour ([topic on meta](https://meta.discourse.org/t/suspend-forever-time-period-is-incendiary/189069)).

This PR special case messages when a user suspended or silenced forever.

### CODE REVIEW

File: app/mailers/user_notifications.rb [hunk 1]
```diff
@@ -153,27 +153,45 @@ def account_silenced(user, opts = nil)
...
     return unless user_history = opts[:user_history]
 
-    build_email(
-      user.email,
-      template: "user_notifications.account_silenced",
-      locale: user_locale(user),
-      reason: user_history.details,
-      silenced_till: I18n.l(user.silenced_till, format: :long)
-    )
+    if user.silenced_till > DateTime.now + 500.years
```
Comment (ZogStriP) [inline app/mailers/user_notifications.rb:156]: Can we instead have a method on the user model like `suspended_forever?` that we can use everywhere?

Comment (AndrewPrigorshnev) [reply app/mailers/user_notifications.rb]: That'll be prefect 👍, done.

File: app/models/user.rb [hunk 2]
```diff
@@ -958,6 +958,10 @@ def silenced_at
     silenced_record.try(:created_at) if silenced?
   end
 
+  def silenced_forever?
+    silenced_till > DateTime.now + 500.years
```
Comment (ZogStriP) [inline app/models/user.rb:962]: Considering the current longevity of website and humans, 100 years should be enough?

```suggestion
    silenced_till > 100.years.from_now
```

Comment (AndrewPrigorshnev) [reply app/models/user.rb]: Hmm, when suspending forever we're suspending even for 1000 years :) But yes 100 years here is better.

**Review by ZogStriP — APPROVED** [review]

File: app/models/user.rb [hunk 3]
```diff
@@ -958,6 +958,10 @@ def silenced_at
     silenced_record.try(:created_at) if silenced?
   end
 
+  def silenced_forever?
+    silenced_till > 100.years.from_now
```
Comment (eviltrout) [inline app/models/user.rb:962]: I hope Discourse lasts long enough that this becomes a bug!

**Review by eviltrout — APPROVED** [review]

**Pull request merged**